scripts/factory-common-install/bin/device-vnc-enter-as-frontend

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+set -euf -o pipefail
+common_factory_install_libexec_dir="$("$(dirname "$0")/pkg-nixos-factory-common-install-get-libexec-dir")"
+. "$common_factory_install_libexec_dir/app_current_device_ssh.sh"
+. "$common_factory_install_libexec_dir/app_current_device_store.sh"
+
+vnc_user=""
+vnc_hostname=""
+
+is_vnc_user_empty() {
+  if [ -z $vnc_user ]; then
+    echo "Error: The vnc user is not set"
+    return 2
+  fi
+}
+
+is_vnc_hostname_empty() {
+  if [ -z $vnc_hostname ]; then
+    echo "Error: The vnc hostname is not set"
+    return 2
+  fi
+}
+
+is_x11vnc_service_stopped() {
+  $(ssh -T ${vnc_user}@${vnc_hostname} systemctl -q --no-pager --user status nixos-secure-factory-x11vnc >/dev/null) && echo ok
+  err=${PIPESTATUS[0]}
+  if [ $err -ne 0 ]; then
+    if [ $err -eq 4 ]; then
+      echo "The service is not started yet, it's all good."
+      return 0
+    else
+      echo "Oops: Something went wrong. Error code: $err"
+      return $err
+    fi
+  fi
+}
+
+cleanup() {
+  echo "Cleanup required"
+  if ! is_x11vnc_service_stopped; then
+    $(ssh -T ${vnc_user}@${vnc_hostname} x11vnc -remote stop) && true
+    err=${PIPESTATUS[0]}
+    if [ $err -ne 0 ]; then
+      echo "error? $err"
+    fi
+  fi
+  exit
+}
+trap cleanup EXIT SIGINT SIGQUIT ERR
+
+enter_vnc_as_user() {
+  vnc_user="$1"
+  local local_port="$2"
+  local remote_port="$3"
+  local err
+  shift 3
+
+  read_or_prompt_for_current_device__hostname "vnc_hostname"
+  read_or_prompt_for_current_device__ssh_port "device_ssh_port"
+
+  if ! is_vnc_user_empty && ! is_vnc_hostname_empty; then
+    echo "Error: VNC user or hostname is not set"
+    exit 2
```

Comment:
use return. Default return for an error is usually 1. Any reason why 2?

Reply:
I chose to return 2 (ENOENT) because at first I thought it was more relevant.. I modified to code to return 1.

Reply:
ok